modules/core/src/main/java/org/apache/fluo/core/client/FluoAdminImpl.java

```diff
@@ -128,19 +140,38 @@ public void initialize(InitializationOptions opts) throws AlreadyInitializedExce
 
     try {
       initialize(conn);
-      updateSharedConfig();
 
-      if (!config.getAccumuloClasspath().trim().isEmpty()) {
-        // TODO add fluo version to context name to make it unique
-        String contextName = "fluo";
+      String accumuloJars;
+      if (!config.getAccumuloJars().trim().isEmpty()) {
+        accumuloJars = config.getAccumuloJars().trim();
+      } else {
+        accumuloJars = getJarsFromClasspath();
+      }
+
+      String accumuloClasspath;
+      if (!accumuloJars.isEmpty()) {
+        accumuloClasspath = copyJarsToHdfs(accumuloJars, "lib/accumulo");
+      } else {
+        accumuloClasspath = config.getAccumuloClasspath().trim();
+      }
+
+      if (!accumuloClasspath.isEmpty()) {
+        String contextName = "fluo-" + config.getApplicationName();
         conn.instanceOperations().setProperty(
-            AccumuloProps.VFS_CONTEXT_CLASSPATH_PROPERTY + "fluo", config.getAccumuloClasspath());
+            AccumuloProps.VFS_CONTEXT_CLASSPATH_PROPERTY + contextName, accumuloClasspath);
         conn.tableOperations().setProperty(config.getAccumuloTable(),
             AccumuloProps.TABLE_CLASSPATH, contextName);
       }
 
+      if (config.getObserverJarsUrl().isEmpty() && !config.getObserverInitDir().trim().isEmpty()) {
```

Comment:
Are there any test for diff combos of these configs?  Combos of accumulo jars, init dir, observers jars, etc